src/Icicle/Dictionary.hs

```diff
@@ -185,14 +204,43 @@ program_count_unique
  = P.Program
  { P.input      = T.IntT
  , P.precomps   = []
- , P.streams    = [(N.Name "inp",  S.Source)]
+ , P.streams    = [(N.Name "inp",  S.Source)
+                  ,(N.Name "inp2", map_fst T.IntT (N.Name "inp"))]
  , P.reduces    = [(N.Name "uniq",
                         R.RFold T.IntT mT
                         (lam mT $ \acc -> lam T.IntT $ \v -> X.XPrim (P.PrimMap $ P.PrimMapInsertOrUpdate T.IntT T.IntT) @~ (lam T.IntT $ \_ -> constI 1) @~ constI 1 @~ v @~ acc)
-                        (X.XPrim (P.PrimConst $ P.PrimConstMapEmpty T.IntT T.IntT))
-                        (N.Name "inp"))]
+                        (X.XValue (T.MapT T.IntT T.IntT) $ N.VMap $ Map.empty)
+                        (N.Name "inp2"))]
+ , P.postdate   = Nothing
  , P.postcomps  = [(N.Name "size", X.XPrim (P.PrimFold (P.PrimFoldMap T.IntT T.IntT) T.IntT) @~ (lam T.IntT $ \a -> lam T.IntT $ \_ -> lam T.IntT $ \b -> a +~ b) @~ constI 0 @~ var "uniq")]
  , P.returns    = var "size"
  }
  where
   mT = T.MapT T.IntT T.IntT
+
+program_days_since_latest :: P.Program Text
```

Comment:
mm no, not really.
maybe I'll try doing some of these
https://github.com/ambiata/ivory/blob/master/ivory-cli/src/test/bin/tests/window/input/dictionary.psv